tests/web_api/test_calculate.py

```diff
@@ -0,0 +1,109 @@
+# -*- coding: utf-8 -*-
+
+import os
+import json
+from httplib import BAD_REQUEST, OK, NOT_FOUND
+
+from nose.tools import assert_equal, assert_in
+import dpath
+
+from . import subject
+
+
+def post_json(data = None, file = None):
+    if file:
+        file_path = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'assets', file)
+        with open(file_path, 'r') as file:
+            data = file.read()
+    return subject.post('/calculate', data = data, content_type = 'application/json')
+
+
+def check_response(data, expected_error_code, path_to_check, content_to_check):
+    response = post_json(data)
+    assert_equal(response.status_code, expected_error_code)
+    json_response = json.loads(response.data)
+    if path_to_check:
+        content = dpath.util.get(json_response, path_to_check)
+        assert_in(content_to_check, content)
+
+
+def test_responses():
+    tests = [
+        ('{"a" : "x", "b"}', BAD_REQUEST, 'error', 'Invalid JSON'),
+        ('["An", "array"]', BAD_REQUEST, 'error', 'Invalid type'),
+        ('{"persons": {}}', BAD_REQUEST, 'persons', 'At least one person'),
+        ('{"persons": {"bob": {}}, "unknown_entity": {}}', BAD_REQUEST, 'unknown_entity', 'entity is not defined',),
```

Comment:
During the workshop on the API, we [agreed](https://github.com/openfisca/openfisca-core/issues/520#issuecomment-308488504) to return a `400` in the case of an unexpected entity.

The main difference is that:
- Unexpected entities rarely happen
- Unexpected variables happen more often, for instance when the user requests a deprecated or renamed variable

We just want a specific error code for the latter, so that we can easily detect it when it happens.